pallets/dca/src/lib.rs

```diff
@@ -430,7 +431,14 @@ pub mod pallet {
 
 			let blocknumber_for_first_schedule_execution = Self::get_next_execution_block(start_execution_block)?;
 
-			Self::plan_schedule_for_block(&who, blocknumber_for_first_schedule_execution, next_schedule_id)?;
+			let mut randomness_generator =
```

Comment:
this leads to the same generator created with every schedule call, therefore every schedule in single block searches for the same blocks. you have to add some salt to the seed unique to the schedule to make sure numbers generated are different, probably `next_schedule_id` could be used

Reply:
alternative solution could be to re-use generator created by dca `on_initialize` and just continue generating number with that one